aspnet/fundamentals/middleware.rst

```diff
@@ -1,8 +1,180 @@
-.. include:: /../common/stub-topic.txt
+Middleware
+==========
+By `Steve Smith`_
 
-|stub-icon| Middleware
-======================
+Small application components that can be incorporated into an HTTP request pipeline are known collectively as middleware. ASP.NET 5 has integrated support for middleware, which are wired up in an application's ``Configure`` method during :doc:`startup`.
 
-.. include:: /../common/stub-notice.txt
+In this article:
+	- `What is middleware`_
+	- `Creating a middleware pipeline with IApplicationBuilder`_
+	- `Built-in middleware`_
+	- `Writing middleware`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/109
+`Download sample from GitHub <https://github.com/aspnet/docs/aspnet/fundamentals/middleware/_static/sample>`_. 
+
+What is middleware
+------------------
+
+Middleware are components that form a pipeline between a server and application for individual requests and responses. Each component can choose whether to pass the request on to the next component in the pipeline, and can perform certain actions before and after the next component in the pipeline. Request delegates are used to build this request pipeline, which are then used to handle each incoming HTTP request to your application. 
```

Comment:
"Middleware are components that are assembled into an application pipeline to handle requests and responses."